packages/commonwealth/client/scripts/views/components/NewThreadForm/NewThreadForm.tsx

```diff
@@ -72,14 +88,21 @@ export const NewThreadForm = () => {
     clearDraft,
   } = useNewThreadForm(chainId, topicsForSelector.enabledTopics);
 
+  const gatedGroupsMatchingTopic = groups?.filter((x) =>
+    x?.topics?.find((y) => y?.id === threadTopic.id),
+  );
+
   const { handleJoinCommunity, JoinCommunityModals } = useJoinCommunity();
   const { isBannerVisible, handleCloseBanner } = useJoinCommunityBanner();
   const { activeAccount: hasJoinedCommunity } = useUserActiveAccount();
 
-  const { data: memberships = [] } = useRefreshMembershipQuery({
-    chainId: app.activeChainId(),
-    address: app?.user?.activeAccount?.address,
-  });
+  const restrictedTopicIds = memberships
```

Comment:
yesterday we updated this implementation and now we can remove this and use this
```
const isTopicGated = !!(memberships || []).find((membership) =>
      membership.topicIds.includes(threadTopic?.id),
    );
    const isActionAllowedInGatedTopic = !!(memberships || []).find(
      (membership) =>
        membership.topicIds.includes(threadTopic?.id) && membership.isAllowed,
    );
    const isRestrictedMembership = isTopicGated && !isActionAllowedInGatedTopic;
```

there is some sample code in `handleNewThreadCreation` of this file